framework/aster-frame/src/task/atomic.rs

```diff
@@ -0,0 +1,208 @@
+        Self { inner }
+    }
+
+    #[must_use]
+    pub fn new_guard(&self) -> AtomicModeGuard {
+        self.inner.inc_cnt();
+        AtomicModeGuard { private: () }
+    }
+
+    pub fn in_atomic_mode(&self) -> bool {
+        self.inner.cnt() != 0
+    }
+
+    fn on_guard_drop(&self) {
+        self.inner.dec_cnt();
+    }
+
+    pub fn info(&self) -> String {
+        format!("ATOMIC_MODE = {}", self.inner.cnt())
+    }
+}
+
+/// The guard of the atomic mode.
+/// The CPU shall exit the atomic mode only after the last guard object is dropped.
+pub struct AtomicModeGuard {
+    #[allow(dead_code)]
+    // This private field prevents user from constructing values of this type directly.
+    private: (),
+}
+
+impl !Send for AtomicModeGuard {}
+
+impl Drop for AtomicModeGuard {
+    fn drop(&mut self) {
+        ATOMIC_MODE.on_guard_drop();
+    }
+}
+
+/// Enter the atomic mode, and return a guard for exiting the current atomic mode.
+/// Nested calls of this method are permitted.
+/// The CPU shall exit the atomic mode only after the last guard object is dropped.
+///
+/// # Example
+///
+/// ```rust
+/// // enters the atomic mode
+/// let guard = enter_atomic_mode();
+/// {
+///     let nested_guard = enter_atomic_mode();
+///     assert!(is_in_atomic_mode());
+///     // do something in atomic mode
+/// }
+/// assert!(is_in_atomic_mode());
+/// drop(guard);    // drop to exit the current atomic mode
+/// assert!(!is_in_atomic_mode());
+/// ```
+#[must_use]
+pub fn enter_atomic_mode() -> AtomicModeGuard {
+    ATOMIC_MODE.new_guard()
+}
+
+/// Whether the current context is in atomic mode.
+pub fn is_in_atomic_mode() -> bool {
+    ATOMIC_MODE.in_atomic_mode()
+}
+
+/// Mark a point after which the code in scope may break the rules of the atomic mode.
+/// Nothing will happen if the current context is not in atomic mode.
+///
+/// # Panic
+///
+/// If the current context is in atomic mode,
+/// a panic will be triggered to indicate the violation of the atomic mode rules.
+pub fn might_break_atomic_mode() {
+    if is_in_atomic_mode() {
+        panic!("Break atomic mode: {}", ATOMIC_MODE.info());
+    }
+}
+
+#[cfg(ktest)]
+mod test {
+    use super::*;
+    mod atomic_mode_inner {
+        use super::*;
+        #[ktest]
+        fn inc_cnt() {
+            let inner = AtomicModeInner::new();
+            inner.inc_cnt();
+            assert_eq!(inner.cnt(), 1);
+        }
+        #[ktest]
+        fn dec_cnt() {
+            let inner = AtomicModeInner::new();
+            inner.inc_cnt();
+            inner.inc_cnt();
+            inner.dec_cnt();
+            assert_eq!(inner.cnt(), 1);
+        }
+    }
+    mod atomic_mode_env {
+        use super::*;
+        #[ktest]
+        fn is_in() {
+            let guard = enter_atomic_mode();
+            assert!(is_in_atomic_mode());
+        }
+        #[ktest]
+        fn is_not_in() {
+            assert!(!is_in_atomic_mode());
+        }
+
```

Comment:
If you ever put empty lines between the two methods `is_not_in` and `entry_multiple_times`, why not put empty lines between `use super::*` and `is_in`, and why not put empty lines between `mod atomic_mode_env` and `mod break_atomic`?

I would suggest that you put empty lines between all methods and all `mod`s. I think this is better than the current format.